[build] CMake build error on Linux, including "src/external/dirent.h" instead of system's

Building with CMake on Linux errors with the following message:
```
/media/dados/prog/raylib/src/external/dirent.h:83:10: fatal error: io.h: No such file or directory
   83 | #include <io.h>         // _findfirst and _findnext set errno iff they return -1
      |          ^~~~~~
```

The build is including `src/external/dirent.h` instead of the system's default `dirent.h`, it seems the problem is adding `src/external` to import path here: https://github.com/raysan5/raylib/blob/master/src/CMakeLists.txt#L108.

--- a/.gitignore
+++ b/.gitignore
@@ -82,7 +82,7 @@ DerivedData/
 
 # Jetbrains project
 .idea/
-cmake-build-debug/
+cmake-build-*/
 
 # CMake stuff
 CMakeCache.txt

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -105,7 +105,6 @@ target_include_directories(raylib
                            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
                            PRIVATE
                            ${CMAKE_CURRENT_SOURCE_DIR}
-                           ${CMAKE_CURRENT_SOURCE_DIR}/external
                            ${CMAKE_BINARY_DIR} # For cmake/config.h
                            ${OPENGL_INCLUDE_DIR}
                            ${OPENAL_INCLUDE_DIR}

